torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2495,7 +2520,8 @@ def _local_pre_load_state_dict_hook(
         state_dict[fqn] = load_tensor
 
     def _sharded_post_load_state_dict_hook(self, *args, **kwargs) -> None:
-        pass
+        if self._use_orig_params:
+            self._register_orig_params()
```

Comment:
Is there guidance for FSDP developers on when they will need to call these methods?

Reply:
I am still working to thoroughly understand how model state dict is implemented, namely the pre/postconditions of the pre/post save and load hooks, e.g. should `FlatParameter`s be registered or should original parameters be registered, what do the prefixes and state dict keys look like at some point in the recursive call stack, etc.

I started trying to retire `FlattenParamsWrapper` but got quickly stymied by trying to understand those pre/postconditions. Maybe after I figure this out, I can help provide more internal documentation around these invariants.